### PR TITLE
feat(logo): animated rainbow background

### PR DESCRIPTION
**Pull request by unseen-ninja**
Cycles through the rainbow-background colours on hover.
Animation only starts on forst hover – animation state persists even after hover status and picks back up again on new hover.

rainbowjack

### CODE REVIEW

File: src/layouts/components/Navigation.astro [hunk 1]
```diff
@@ -38,6 +38,14 @@ import { navigationLinks } from "@data/links";
...
     width: 4rem;
     height: 4rem;
+
+    :global(svg *) {
+      animation-play-state: paused;
+    }
+
+    &:hover :global(svg *) {
+      animation-play-state: running;
+    }
```
Comment (sgoudham) [inline src/layouts/components/Navigation.astro:48]: We should probably do ourselves a favour going forward to _always_ document the reasoning for a `:global` style.

Comment (unseen-ninja) [reply src/layouts/components/Navigation.astro:48]: It's mostly due to some quirks with how Astro does the CSS scoping. At the end of the day it just needs to boil down to `.nav-logo svg *` to function. If we find a better way to achieve that I'd be stoked to use that instead.

Happy to add a comment about it into the file itself. c:

**Comment by unseen-ninja** [comment]
May the `-f` be with you..

**Review by sgoudham — APPROVED** [review]
A really nice touch, thanks!

**Pull request merged**